math/binary_exponent.cpp

```diff
@@ -50,22 +96,39 @@ int binExpo_alt(int a, int b) {
     }
     return res;
 }
+}  // namespace math
 
-/// Main function
-int main() {
-    int a, b;
-    /// Give two numbers a, b
-    std::cin >> a >> b;
-    if (a == 0 && b == 0) {
-        std::cout << "Math error" << std::endl;
-    } else if (b < 0) {
-        std::cout << "Exponent must be positive !!" << std::endl;
-    } else {
-        int resRecurse = binExpo(a, b);
-        /// int resIterate = binExpo_alt(a, b);
+using namespace math;
```

Comment:
It doesn't harm if you use `math` directly. This looks unclean and unprofessional, IMO.

Reply:
can you expand on unprofessional? I'm assuming unclean is because using's are meant to be at the top of the file ( can't remember why I put it on this line ). I can't seem to see another issue; the intention I had was to not use a lot of scope resolution operators, making it a little easier to read.

Reply:
Scope resolution is a good thing for a few reasons and I don't think I need to go into the details here, but I would just give you two reasons why `using namespace abc` is a bad idea:

1. Name collisions - the main reason namespacing was created, consider that you are using the classic `using namespace std;` and you created a `string` class in your current `namespace abc`, `using namespace std;` would cause a naming collision which actually is very difficult to resolve when you have a big dependency chain (in small scale programs you won't notice any problems but once you get to bigger projects you would start to dislike people who use it in their code 😂 jkjk)
2. Debugging problems - this also applies to bigger codebases, the `using namespace abc` would make your life very difficult when using `gdb` or any other debugging tool (that is not paid for).

For the matter of "unprofessional", I think it shows that the coder is novice so that's a different thing from unprofessional but still should be relevant 😄